src/bootstrap/compile.rs

```diff
@@ -161,14 +161,15 @@ impl Step for Std {
         // The LLD wrappers and `rust-lld` are self-contained linking components that can be
         // necessary to link the stdlib on some targets. We'll also need to copy these binaries to
         // the `stage0-sysroot` to ensure the linker is found when bootstrapping on such a target.
-        if compiler.stage == 0 && compiler.host == builder.config.build {
+        let src_sysroot_bin = builder
+            .rustc_snapshot_sysroot()
+            .join("lib")
+            .join("rustlib")
+            .join(compiler.host.triple)
+            .join("bin");
+        if compiler.stage == 0 && compiler.host == builder.config.build && src_sysroot_bin.exists()
+        {
```

Comment:
Nit Change: We don't need to traverse the `src_sysroot_bin` path and check its existence when it's not necessary.

like:

```rs
if compiler.stage == 0 && compiler.host == builder.config.build {
    let src_sysroot_bin = ....
    
    if src_sysroot_bin.exists() {
       ....
    }
}
```

Reply:
Same, and the path creation code would then also stay with its dedicated comment.